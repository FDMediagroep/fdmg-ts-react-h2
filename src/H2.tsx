import * as React from 'react';

export interface Props {
    className?: string;
    text: string;
}

/**
 * H2 compononent renders a heading
 */
export default class H2 extends React.Component<Props, any> {
    public state: any;
    public props: Props;

    constructor(props: Props) {
        super(props);
        this.props = props;
        console.info('H2 component', props);
    }

    componentDidMount() {
        console.info('H2 mounted');
    }
        
    componentWillUnmount() {
        console.info('H2 will unmount');
    }

    componentWillReceiveProps(nextProps) {
        console.info('H2 next props', nextProps);
    }

    render() {
        return (
            <h2 className={this.props.className}>{this.props.text}</h2>
        );
    }
}
