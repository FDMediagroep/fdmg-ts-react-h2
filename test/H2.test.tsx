import * as React from 'react';
import * as renderer from 'react-test-renderer';
import H2 from '../src/H2';

console.info = () => {};

describe('H2', () => {
    test('renders correctly', () => {
        let component = renderer.create(
            <H2/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        component = renderer.create(
            <H2>test h2</H2>
        );
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        component = renderer.create(
            <H2><span>testing 123</span><div>testing 456</div></H2>
        );
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('renders correctly with CSS class', () => {
        const component = renderer.create(
            <H2 className={'title'}>test h2</H2>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
