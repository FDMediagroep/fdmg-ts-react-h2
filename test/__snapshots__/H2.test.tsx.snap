// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H2 renders correctly 1`] = `
<h2
  className={undefined}
/>
`;

exports[`H2 renders correctly 2`] = `
<h2
  className={undefined}
/>
`;

exports[`H2 renders correctly 3`] = `
<h2
  className={undefined}
/>
`;

exports[`H2 renders correctly with CSS class 1`] = `
<h2
  className="title"
>
  test h2
</h2>
`;
